app/models/point.rb

```diff
@@ -0,0 +1,7 @@
+class Point < ApplicationRecord
+  validates :x, presence: true
+  validates :y, presence: true
+
+  has_many :point_of_interests, dependent: :destroy
+  has_and_belongs_to_many :polylines
```

Comment:
codefactor doesn't like the has_and_belongs_to_many. As Romeo already researched on https://flatironschool.com/blog/why-you-dont-need-has-and-belongs-to-many we could try a has_many :through relation. Would any technical aspect play against this?

Reply:
https://guides.rubyonrails.org/association_basics.html#choosing-between-has-many-through-and-has-and-belongs-to-many Here, it says using `has_many through` is recommended when using the relationship model (e.g PolylinesPoints) as an independent entity. We prefer to leave the join models on 

- PointsPolylines
- PolylinesWalls
- RoomsWalls
- RoomsPointsOfInterest

 implicit (because we don't need to work on these models independently). I opened a pull request for adjusting this rule (https://github.com/hpi-swt2/compass-portal-red/pull/54)